operations/commit_queue.go

```diff
@@ -86,6 +94,59 @@ func deleteItem() cli.Command {
 	}
 }
 
+func mergeCommand() cli.Command {
+	return cli.Command{
+		Name:  "merge",
+		Usage: "test and merge a feature branch",
+		Flags: mergeFlagSlices(addProjectFlag(), addLargeFlag(), addYesFlag(
+			cli.StringFlag{
+				Name:  joinFlagNames(refFlagName, "r"),
+				Usage: "merge branch `REF`",
+				Value: "HEAD",
+			},
+			cli.StringFlag{
+				Name:  identifierFlagName,
+				Usage: "finalize a preexisting item with `ID`",
+			},
+			cli.BoolFlag{
+				Name:  pauseFlagName,
+				Usage: "wait to enqueue an item until finalized",
+			},
+			cli.StringFlag{
+				Name:  joinFlagNames(patchDescriptionFlagName, "d"),
+				Usage: "description for the patch",
+			},
+		)),
+		Before: mergeBeforeFuncs(
+			setPlainLogger,
+		),
+		Action: func(c *cli.Context) error {
+			confPath := c.Parent().String(confFlagName)
+			projectID := c.String(projectFlagName)
+			ref := c.String(refFlagName)
+			id := c.String(identifierFlagName)
+			pause := c.Bool(pauseFlagName)
+			largeOK := c.Bool(largeFlagName)
+			description := c.String(patchDescriptionFlagName)
+			skipConfirm := c.Bool(yesFlagName)
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			params := mergeParams{
+				projectID:   projectID,
+				ref:         ref,
+				id:          id,
+				pause:       pause,
+				description: description,
+				skipConfirm: skipConfirm,
+				large:       largeOK,
+			}
+			return params.mergeBranch(ctx, confPath)
```

Comment:
all the logic is removed from the Action (looking at Sam...)

Reply:
sweet! I think it'd be fine to populate mergeParams{} directly from calls to methods on the cli.Context

Reply:
done.